internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -3519,7 +3519,42 @@ var _ = Describe("selectClusterNetworkType", func() {
 	}
 })
 
-var _ = Describe("Getting ClusterDeployment admin kubeconfig secret name", func() {
+var _ = Describe("Getting spoke admin kubeconfig secret name from ClusterReference", func() {
+	var (
+		clusterName             = "test-cluster"
+		agentClusterInstallName = "test-cluster-aci"
+		pullSecretName          = "pull-secret"
+		adminSecretName         = "admin-secret"
+		clusterReference        = aiv1beta1.ClusterReference{Name: clusterName, Namespace: testNamespace}
+		ctx                     = context.Background()
+		c                       client.Client
+		defaultClusterSpec      hivev1.ClusterDeploymentSpec
+		cd                      *hivev1.ClusterDeployment
+	)
+	BeforeEach(func() {
+		c = fakeclient.NewClientBuilder().WithScheme(scheme.Scheme).Build()
+		cd = newClusterDeployment(clusterName, testNamespace, defaultClusterSpec)
+		cd.Spec.ClusterMetadata = &hivev1.ClusterMetadata{
+			AdminKubeconfigSecretRef: corev1.LocalObjectReference{
+				Name: adminSecretName,
+			},
+		}
+	})
+	Context("ClusterDeployment does not exist", func() {
+		It("can make secret name from template", func() {
+			Expect(getClusterReferenceAdminKubeConfigSecretName(ctx, c, &clusterReference)).To(Equal(fmt.Sprintf(adminKubeConfigStringTemplate, cd.Name)))
+		})
+	})
+	Context("ClusterDeployment exists", func() {
+		It("can extract secret name from ClusterDeployment", func() {
```

Comment:
Same here, this should not have a context and be named something like:
`It("gets the secret name from the cluster deployment when it exists", func() {`